src/Core/OrganizationFeatures/OrganizationUsers/AcceptOrgUserCommand.cs

```diff
@@ -0,0 +1,224 @@
+﻿using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.OrganizationFeatures.OrganizationUsers.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationUsers;
+
+public class AcceptOrgUserCommand : IAcceptOrgUserCommand
+{
+    private readonly IDataProtector _dataProtector;
+    private readonly IGlobalSettings _globalSettings;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IPolicyService _policyService;
+    private readonly IMailService _mailService;
+    private readonly IUserRepository _userRepository;
+    private readonly IDataProtectorTokenFactory<OrgUserInviteTokenable> _orgUserInviteTokenDataFactory;
+
+    public AcceptOrgUserCommand(
+        IDataProtectionProvider dataProtectionProvider,
+        IGlobalSettings globalSettings,
+        IOrganizationUserRepository organizationUserRepository,
+        IOrganizationRepository organizationRepository,
+        IPolicyService policyService,
+        IMailService mailService,
+        IUserRepository userRepository,
+        IDataProtectorTokenFactory<OrgUserInviteTokenable> orgUserInviteTokenDataFactory)
+    {
+
+        // TODO: remove data protector when old token validation removed
+        _dataProtector = dataProtectionProvider.CreateProtector("OrganizationServiceDataProtector");
+        _globalSettings = globalSettings;
+        _organizationUserRepository = organizationUserRepository;
+        _organizationRepository = organizationRepository;
+        _policyService = policyService;
+        _mailService = mailService;
+        _userRepository = userRepository;
+        _orgUserInviteTokenDataFactory = orgUserInviteTokenDataFactory;
+    }
+
+    public async Task<OrganizationUser> AcceptOrgUserByEmailTokenAsync(Guid organizationUserId, User user, string emailToken,
+        IUserService userService)
```

Comment:
suggestion: I know its from the `OrganizationService`, but requiring the `IUserService` as an argument here seems like an anti-pattern. Do you think there's anyway we can refactor this dependency? It looks like its only being used internally to check if the user has some form of 2FA enabled. 

If its too much/tricky to try and include in this work then no worries, I don't think it should be a blocker as its already what exists in the service.

Reply:
Yeah, we recognized it was an anti-pattern while moving the code, but didn't want to expand the scope to also include refactoring that considering how much I did refactor.  However, creating a tech debt for it was a good call. I've done so with https://bitwarden.atlassian.net/browse/PM-4320. Thank you!